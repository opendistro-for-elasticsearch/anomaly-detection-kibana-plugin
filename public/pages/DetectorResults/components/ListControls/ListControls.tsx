/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem, EuiPagination } from '@elastic/eui';

interface ListControlsProps {
  activePage: number;
  pageCount: number;
  onPageClick: (pageNumber: number) => void;
}
export const ListControls = (props: ListControlsProps) => (
  <EuiFlexGroup
    style={{ padding: '0px 5px' }}
    alignItems="center"
    justifyContent="flexEnd"
  >
    {props.pageCount > 1 ? (
      <EuiFlexItem grow={false} style={{ justifyContent: 'center' }}>
        <EuiPagination
          pageCount={props.pageCount}
          activePage={props.activePage}
          onPageClick={props.onPageClick}
          data-test-subj="anomaliesPageControls"
        />
      </EuiFlexItem>
    ) : null}
  </EuiFlexGroup>
);
