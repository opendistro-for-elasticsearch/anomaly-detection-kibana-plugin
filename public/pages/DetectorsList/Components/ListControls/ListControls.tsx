/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiComboBox,
  EuiComboBoxOptionProps,
  EuiFieldSearch,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPagination,
} from '@elastic/eui';
import React from 'react';
import { getDetectorStateOptions } from '../../utils/helpers';
import { DETECTOR_STATE } from '../../../../utils/constants';

interface ListControlsProps {
  activePage: number;
  pageCount: number;
  search: string;
  selectedDetectorStates: DETECTOR_STATE[];
  selectedIndices: string[];
  indexOptions: EuiComboBoxOptionProps[];
  onDetectorStateChange: (options: EuiComboBoxOptionProps[]) => void;
  onIndexChange: (options: EuiComboBoxOptionProps[]) => void;
  onSearchDetectorChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onSearchIndexChange: (searchValue: string) => void;
  onPageClick: (pageNumber: number) => void;
}
export const ListControls = (props: ListControlsProps) => (
  <EuiFlexGroup gutterSize="s">
    <EuiFlexItem grow={false} style={{ width: '40%' }}>
      <EuiFieldSearch
        fullWidth={true}
        value={props.search}
        placeholder="Search"
        onChange={props.onSearchDetectorChange}
        data-test-subj="detectorListSearch"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiComboBox
        id="selectedDetectorStates"
        placeholder="All detector states"
        isClearable={true}
        singleSelection={false}
        options={getDetectorStateOptions()}
        onChange={props.onDetectorStateChange}
        selectedOptions={
          props.selectedDetectorStates.length > 0
            ? props.selectedDetectorStates.map(index => ({ label: index }))
            : []
        }
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiComboBox
        id="selectedIndices"
        placeholder="All indices"
        isClearable={true}
        singleSelection={false}
        options={props.indexOptions}
        onChange={props.onIndexChange}
        onSearchChange={props.onSearchIndexChange}
        selectedOptions={
          props.selectedIndices.length > 0
            ? props.selectedIndices.map(index => ({ label: index }))
            : []
        }
      />
    </EuiFlexItem>
    {props.pageCount > 1 ? (
      <EuiFlexItem grow={false} style={{ justifyContent: 'center' }}>
        <EuiPagination
          pageCount={props.pageCount}
          activePage={props.activePage}
          onPageClick={props.onPageClick}
          data-test-subj="detectorPageControls"
        />
      </EuiFlexItem>
    ) : null}
  </EuiFlexGroup>
);
