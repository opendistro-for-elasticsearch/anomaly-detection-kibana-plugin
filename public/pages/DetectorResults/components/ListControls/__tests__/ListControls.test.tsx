/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ListControls } from '../ListControls';

describe('<ListControls /> spec', () => {
  const defaultProps = {
    activePage: 1,
    pageCount: 10,
    search: '',
    onSearchChange: jest.fn(),
    onPageClick: jest.fn(),
  };
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('Empty results', () => {
    test('renders component with empty message', async () => {
      const { container } = render(<ListControls {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    test('pagination should be visible if pages count more than 1', async () => {
      const { queryByTestId } = render(
        <ListControls {...defaultProps} pageCount={2} />
      );
      expect(queryByTestId('anomaliesPageControls')).not.toBeNull();
    });
    test('should call onPageClick when click on the page', async () => {
      const { getByTestId } = render(<ListControls {...defaultProps} />);
      fireEvent.click(getByTestId('pagination-button-3'));
      expect(defaultProps.onPageClick).toHaveBeenCalledTimes(1);
    });
  });
});
