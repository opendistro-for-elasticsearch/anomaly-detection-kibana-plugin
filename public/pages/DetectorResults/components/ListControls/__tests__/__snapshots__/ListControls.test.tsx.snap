// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListControls /> spec Empty results renders component with empty message 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding: 0px 5px;"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <div
      class="euiPagination"
      data-test-subj="anomaliesPageControls"
      role="group"
    >
      <button
        aria-label="Previous page"
        class="euiButtonIcon euiButtonIcon--text"
        data-test-subj="pagination-button-previous"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
      <button
        aria-label="Page 1 of 5"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
        data-test-subj="pagination-button-0"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            1
          </span>
        </span>
      </button>
      <button
        aria-label="Page 2 of 5"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
        data-test-subj="pagination-button-1"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            2
          </span>
        </span>
      </button>
      <button
        aria-label="Page 3 of 5"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
        data-test-subj="pagination-button-2"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            3
          </span>
        </span>
      </button>
      <button
        aria-label="Page 4 of 5"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
        data-test-subj="pagination-button-3"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            4
          </span>
        </span>
      </button>
      <button
        aria-label="Page 5 of 5"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
        data-test-subj="pagination-button-4"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            5
          </span>
        </span>
      </button>
      <button
        aria-hidden="true"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton-isPlaceholder euiPaginationButton--hideOnMobile"
        disabled=""
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            <span>
              â€¦
            </span>
          </span>
        </span>
      </button>
      <button
        aria-label="Jump to the last page, number 10"
        class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiPaginationButton euiPaginationButton--hideOnMobile"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            10
          </span>
        </span>
      </button>
      <button
        aria-label="Next page"
        class="euiButtonIcon euiButtonIcon--text"
        data-test-subj="pagination-button-next"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
    </div>
  </div>
</div>
`;
